src/components/molecules/ScheduleEvent/ScheduleEvent.tsx

```diff
@@ -81,26 +122,103 @@ export const ScheduleEvent: React.FC<ScheduleEventProps> = ({
 
   const onEventBoxClick = useCallback(
     (e) => {
-      if (e.target.closest(`.${ScheduleEventBookmarkClass}`)) return;
+      if (
+        e.target.closest(`.${ScheduleEventBookmarkClass}`) ||
+        e.target.closest(`.${ScheduleEventExpandClass}`)
+      )
+        return;
 
       showEventModal();
     },
     [showEventModal]
   );
 
+  const handleMouseOver = useCallback(() => {
+    if (isMinimallyLongEvent) return;
+
+    showExpanded();
+  }, [isMinimallyLongEvent, showExpanded]);
+
+  const handleMouseOut = useCallback(() => {
+    if (isExpandedStatic) return;
+
+    hideExpanded();
+  }, [isExpandedStatic, hideExpanded]);
+
+  const handleCollapse = useCallback(() => {
+    hideExpandedStatic();
+    hideExpanded();
+  }, [hideExpanded, hideExpandedStatic]);
+
+  const handleToggleExpand = useCallback(() => {
+    if (!isExpandedStatic) {
+      toggleExpandedStatic();
+
+      return;
+    }
+
+    handleCollapse();
+  }, [isExpandedStatic, handleCollapse, toggleExpandedStatic]);
+
+  useEffect(() => {
+    const handleClickOutside = (event: Event) => {
+      const eventTarget = event.target as Node;
```

Comment:
Never use `as`, it effectively nullifies any type safety we get from TypeScript, and hides issues that would be caught at dev time and makes them runtime issues in production. There is practically if not literally always a better way to solve the need than by using `as`.

Reply:
done f718762